In [1]:
import pandas as pd
import numpy as np
import sqlite3



### Loading in the most backed projects data from kickstarter

In [3]:
most_backed = pd.read_csv('most_backed.csv')
most_backed.head()

Unnamed: 0  amt.pledged                                              blurb  \
0           0    8782571.0  \nThis is a card game for people who are into ...   
1           1    6465690.0  \nAn unusually addicting, high-quality desk to...   
2           2    5408916.0  \nBring Reading Rainbow’s library of interacti...   
3           3    5702153.0  \nUPDATED: This is it. We're making a Veronica...   
4           4    3336371.0  \nAn adventure game from Tim Schafer, Double F...   

                                     by        category currency       goal  \
0                              Elan Lee  Tabletop Games      usd    10000.0   
1            Matthew and Mark McLachlan  Product Design      usd    15000.0   
2        LeVar Burton & Reading Rainbow             Web      usd  1000000.0   
3                            Rob Thomas  Narrative Film      usd  2000000.0   
4  Double Fine and 2 Player Productions     Video Games      usd   400000.0   

            location  num.backers  \
0    Los Angeles, CA       219382   
1         Denver, CO       154926   
2    Los Angeles, CA       105857   
3      San Diego, CA        91585   
4  San Francisco, CA        87142   

                                    num.backers.tier  \
0                            [15505, 202934, 200, 5]   
1  [788, 250, 43073, 21796, 41727, 21627, 12215, ...   
2  [19639, 14343, 9136, 2259, 5666, 24512, 4957, ...   
3  [5938, 8423, 11509, 22997, 23227, 1865, 7260, ...   
4  [47946, 24636, 1090, 11530, 900, 148, 100, 10, 4]   

                                         pledge.tier  \
0                         [20.0, 35.0, 100.0, 500.0]   
1  [1.0, 14.0, 19.0, 19.0, 35.0, 35.0, 79.0, 79.0...   
2  [5.0, 10.0, 25.0, 30.0, 35.0, 50.0, 75.0, 100....   
3  [1.0, 10.0, 25.0, 35.0, 50.0, 75.0, 100.0, 150...   
4  [15.0, 30.0, 60.0, 100.0, 250.0, 500.0, 1000.0...   

                                               title  \
0                                  Exploding Kittens   
1                      Fidget Cube: A Vinyl Desk Toy   
2  Bring Reading Rainbow Back for Every Child, Ev...   
3                    The Veronica Mars Movie Project   
4                              Double Fine Adventure   

                                                 url  
0                /projects/elanlee/exploding-kittens  
1   /projects/antsylabs/fidget-cube-a-vinyl-desk-toy  
2  /projects/readingrainbow/bring-reading-rainbow...  
3  /projects/559914737/the-veronica-mars-movie-pr...  
4         /projects/doublefine/double-fine-adventure

# Auditing accuracy
In the data shown above, we see the 5 most backed projects on kickstater. We can see that the data holds true by going to kickstarters website and listing the most backed projects on their site.
https://www.kickstarter.com/discover/advanced?woe_id=23424977&sort=most_backed&seed=2528755&page=3

# Auditing completeness

Comparing column metrics of our dataset with a similar Kickstarter project dataset on Kaggle, we find a few columns missing in our dataset as well as a few extra columns of information compared to the dataset found at https://www.kaggle.com/kemical/kickstarter-projects/data
The column metadata of the said dataset is as shown below:

<img src = 'kickstarter.png'>

#### Comparing these column attributes to the attributes of our dataset:

In [252]:
most_backed.columns

Index(['Project ID', 'Amount Pledged', 'Blurb', 'Inventor ID', 'Category ID',
       'Currency', 'Goal', 'Location', 'No. Backers', 'Title', 'URL'],
      dtype='object')

#### The attributes missing in our dataset are: <br>
1 A subcategory column in addition to the category column <br>
2 The launch data and end date of the project <br>
3 State of the project (Success or failure)<br>
<br>
#### The extra attributes in our dataset are: <br>
1 Information on the various tiers a backer can pledge (20, 40, 60 etc) on a specific project <br>
2 The number of backers for each pledge tier <br>
3 URL of each project <br>
4 Blurb for each project <br>
5 The name of the creator for each project <br>
Overall we can say that our dataset is more complete and analaysis-ready than other datasets found on kaggle. <br>

# Auditing consistency/uniformity

In [5]:
most_backed.describe()

Unnamed: 0   amt.pledged          goal    num.backers
count  4000.000000  4.000000e+03  4.000000e+03    4000.000000
mean   1999.500000  2.899923e+05  6.175203e+04    3582.875750
std    1154.844867  7.119369e+05  1.266715e+05    7316.102809
min       0.000000  1.925000e+03  1.000000e+00    1109.000000
25%     999.750000  6.771300e+04  1.300000e+04    1406.750000
50%    1999.500000  1.205860e+05  3.000000e+04    1946.000000
75%    2999.250000  2.593680e+05  6.000000e+04    3372.000000
max    3999.000000  2.033899e+07  2.000000e+06  219382.000000

In [7]:
most_backed.isnull().sum()

Unnamed: 0          0
amt.pledged         0
blurb               0
by                  0
category            0
currency            0
goal                0
location            0
num.backers         0
num.backers.tier    0
pledge.tier         0
title               0
url                 0
dtype: int64

### We see that there are no null values in our dataset. Our dataset appears to be pre-cleaned and ready for analysis. We also see that the data appears to be consistent throughout the rows and the possibility of outliers in the dataset is low.

In [8]:
conn = sqlite3.connect('kickstarter_prokects.db')

In [10]:
most_backed.to_sql('most_backed',conn, index = False, if_exists='replace')

C:\Users\hashe\Anaconda3\lib\site-packages\pandas\core\generic.py:1362: UserWarning: The spaces in these column names will not be changed. In pandas versions < 0.14, spaces were converted to underscores.
  chunksize=chunksize, dtype=dtype)


In [18]:
def run_query(query):
    return pd.read_sql(query,conn)

In [19]:
most_backed.head()

Unnamed: 0  amt.pledged                                              blurb  \
0           0    8782571.0  \nThis is a card game for people who are into ...   
1           1    6465690.0  \nAn unusually addicting, high-quality desk to...   
2           2    5408916.0  \nBring Reading Rainbow’s library of interacti...   
3           3    5702153.0  \nUPDATED: This is it. We're making a Veronica...   
4           4    3336371.0  \nAn adventure game from Tim Schafer, Double F...   

                                     by        category currency       goal  \
0                              Elan Lee  Tabletop Games      usd    10000.0   
1            Matthew and Mark McLachlan  Product Design      usd    15000.0   
2        LeVar Burton & Reading Rainbow             Web      usd  1000000.0   
3                            Rob Thomas  Narrative Film      usd  2000000.0   
4  Double Fine and 2 Player Productions     Video Games      usd   400000.0   

            location  num.backers  \
0    Los Angeles, CA       219382   
1         Denver, CO       154926   
2    Los Angeles, CA       105857   
3      San Diego, CA        91585   
4  San Francisco, CA        87142   

                                    num.backers.tier  \
0                            [15505, 202934, 200, 5]   
1  [788, 250, 43073, 21796, 41727, 21627, 12215, ...   
2  [19639, 14343, 9136, 2259, 5666, 24512, 4957, ...   
3  [5938, 8423, 11509, 22997, 23227, 1865, 7260, ...   
4  [47946, 24636, 1090, 11530, 900, 148, 100, 10, 4]   

                                         pledge.tier  \
0                         [20.0, 35.0, 100.0, 500.0]   
1  [1.0, 14.0, 19.0, 19.0, 35.0, 35.0, 79.0, 79.0...   
2  [5.0, 10.0, 25.0, 30.0, 35.0, 50.0, 75.0, 100....   
3  [1.0, 10.0, 25.0, 35.0, 50.0, 75.0, 100.0, 150...   
4  [15.0, 30.0, 60.0, 100.0, 250.0, 500.0, 1000.0...   

                                               title  \
0                                  Exploding Kittens   
1                      Fidget Cube: A Vinyl Desk Toy   
2  Bring Reading Rainbow Back for Every Child, Ev...   
3                    The Veronica Mars Movie Project   
4                              Double Fine Adventure   

                                                 url  
0                /projects/elanlee/exploding-kittens  
1   /projects/antsylabs/fidget-cube-a-vinyl-desk-toy  
2  /projects/readingrainbow/bring-reading-rainbow...  
3  /projects/559914737/the-veronica-mars-movie-pr...  
4         /projects/doublefine/double-fine-adventure

#### Creating a list of the pledge tiers and its associated backers

In [20]:
pledge_list=[]
for string in most_backed['pledge.tier']:
    ind_lis=[]
    string = string.replace('[','').replace(']','').replace(' ...','').split(',')
    for strin in string:
        if strin != '':
            ind_lis.append(float(strin))
    pledge_list.append(ind_lis)

In [21]:
backers_list=[]
for string in most_backed['num.backers.tier']:
    ind_lis=[]
    string = string.replace('[','').replace(']','').replace(' ...','').split(',')
    for strin in string:
        if strin != '':
            ind_lis.append(float(strin))
    backers_list.append(ind_lis)

### Creating a dataframe of the pledge tiers and backers. 
(I have split the process into 4 parts each as Jupyter notebook seems to run into problems when consolidating all 4000 rows at once)

In [59]:
frames=[]
for each in pledge_list[:1000]:
    df = pd.DataFrame(each, columns = ['Pledge Tiers'])
    frames.append(df)
    pledge_df_1= pd.concat(frames)

In [61]:
frames=[]
for each in pledge_list[1000:2000]:
    df = pd.DataFrame(each, columns = ['Pledge Tiers'])
    frames.append(df)
    pledge_df_2= pd.concat(frames)

In [66]:
frames=[]
for each in pledge_list[2000:3000]:
    df = pd.DataFrame(each, columns = ['Pledge Tiers'])
    frames.append(df)
    pledge_df_3= pd.concat(frames)

In [63]:
frames=[]
for each in pledge_list[3000:]:
    df = pd.DataFrame(each, columns = ['Pledge Tiers'])
    frames.append(df)
    pledge_df_4= pd.concat(frames)

In [69]:
pledge_df = pd.concat([pledge_df_1, pledge_df_2, pledge_df_3, pledge_df_4])

In [70]:
pledge_df.shape

(57990, 1)

In [95]:
frames=[]
for each in backers_list[:1000]:
    df = pd.DataFrame(each, columns = ['Backers'])
    frames.append(df)
    backers_df_1= pd.concat(frames)

In [97]:
frames=[]
for each in backers_list[1000:2000]:
    df = pd.DataFrame(each, columns = ['Backers'])
    frames.append(df)
    backers_df_2= pd.concat(frames)

In [99]:
frames=[]
for each in backers_list[2000:3000]:
    df = pd.DataFrame(each, columns = ['Backers'])
    frames.append(df)
    backers_df_3= pd.concat(frames)

In [101]:
frames=[]
for each in backers_list[3000:4000]:
    df = pd.DataFrame(each, columns = ['Backers'])
    frames.append(df)
    backers_df_4= pd.concat(frames)

In [103]:
backers_df = pd.concat([backers_df_1,backers_df_2,backers_df_3,backers_df_4])

In [104]:
backers_df.shape

(57990, 1)

In [123]:
pledge_tiers_df = pd.concat([pledge_df, backers_df], axis = 1)

In [143]:
pledge_tiers_df = pledge_tiers_df[['Pledge Tiers', 'Backers']]
pledge_tiers_df['Pledge ID'] = 1
pledge_tiers_df.reset_index(drop=True, inplace = True)

In [144]:
count = 0
for i, value in enumerate(pledge_list):
    for each in value:
        pledge_tiers_df.loc[count, 'Pledge ID'] = i
        count+=1

In [145]:
pledge_tiers_df = pledge_tiers_df[['Pledge ID', 'Pledge Tiers', 'Backers']]
pledge_tiers_df.head()

Pledge ID  Pledge Tiers   Backers
0          0          20.0   15505.0
1          0          35.0  202934.0
2          0         100.0     200.0
3          0         500.0       5.0
4          1           1.0     788.0

In [146]:
pledge_tiers_df.to_sql('Pledge_Tiers',conn,index= False, if_exists = 'replace')

C:\Users\hashe\Anaconda3\lib\site-packages\pandas\core\generic.py:1362: UserWarning: The spaces in these column names will not be changed. In pandas versions < 0.14, spaces were converted to underscores.
  chunksize=chunksize, dtype=dtype)


In [150]:
run_query('select name from sqlite_master where type = "table"')

name
0   most_backed
1  Pledge_Tiers

In [173]:
inventor_list = list(set(most_backed_df['by']))
inventor_list[1514]

'LeVar Burton & Reading Rainbow'

In [202]:
most_backed.columns = ['Project ID', 'Amount Pledged', 'Blurb', 'Inventor ID', 'Category ID', 'Currency', 'Goal', 'Location', 'No. Backers', 'Tier', 'P tier', 'Title', 'URL']

In [187]:
category_list = list(set(most_backed_df['category']))
category_list[101]

'Product Design'

In [183]:
for i, value in enumerate(most_backed['category']):
    for a, val in enumerate(category_list):
        if value == val:
            most_backed.loc[i,'category'] = a

In [208]:
most_backed = most_backed[['Project ID', 'Amount Pledged', 'Blurb', 'Inventor ID', 'Category ID', 'Currency', 'Goal', 'Location', 'No. Backers', 'Title', 'URL']]

In [209]:
most_backed

Project ID  Amount Pledged  \
0              0       8782571.0   
1              1       6465690.0   
2              2       5408916.0   
3              3       5702153.0   
4              4       3336371.0   
5              5      20338986.0   
6              6       4188927.0   
7              7       3986929.0   
8              8       2090104.0   
9              9       3007370.0   
10            10       6333295.0   
11            11      10266845.0   
12            12       3845170.0   
13            13      12779843.0   
14            14       5545991.0   
15            15       3246588.0   
16            16       8596474.0   
17            17      13285226.0   
18            18       2933252.0   
19            19       5764229.0   
20            20       3105473.0   
21            21       9192055.0   
22            22       2229344.0   
23            23       2032434.0   
24            24       2785537.0   
25            25       2071927.0   
26            26       1731465.0   
27            27       1836447.0   
28            28       1106371.0   
29            29       1479046.0   
...          ...             ...   
3970        3970         16692.0   
3971        3971        100348.0   
3972        3972         30303.0   
3973        3973        122264.0   
3974        3974        105147.0   
3975        3975        155838.0   
3976        3976        378725.0   
3977        3977         42901.0   
3978        3978         82811.0   
3979        3979         51213.0   
3980        3980         91465.0   
3981        3981         31107.0   
3982        3982         37622.0   
3983        3983         82056.0   
3984        3984        108893.0   
3985        3985         40356.0   
3986        3986         69829.0   
3987        3987         67767.0   
3988        3988         47450.0   
3989        3989         47321.0   
3990        3990         17570.0   
3991        3991        147396.0   
3992        3992        145034.0   
3993        3993        467167.0   
3994        3994         46982.0   
3995        3995         93331.0   
3996        3996        186144.0   
3997        3997         60955.0   
3998        3998         69452.0   
3999        3999         24409.0   

                                                  Blurb Inventor ID  \
0     \nThis is a card game for people who are into ...         930   
1     \nAn unusually addicting, high-quality desk to...          92   
2     \nBring Reading Rainbow’s library of interacti...        1514   
3     \nUPDATED: This is it. We're making a Veronica...        1928   
4     \nAn adventure game from Tim Schafer, Double F...         627   
5     \nColor e-paper smartwatch with up to 7 days o...        1709   
6     \nA story-driven CRPG set in the world of Mont...        2916   
7     \nPillars of Eternity is an isometric, party-b...         926   
8     \nThe creative talent behind Banjo-Kazooie and...         562   
9     \nZNAPS is a magnetic adapter specially design...        1730   
10    \nYu Suzuki presents the long awaited third in...        2263   
11    \nPebble is a customizable watch. Download new...        1709   
12    \nClassic Japanese side-scrolling action, evol...         237   
13    \nTwo affordable, heart rate-enabled smartwatc...        1709   
14    \nIGAVANIA (n): A gothic, exploration-focused ...        2548   
15    \nFrom the creators of Cyanide & Happiness com...        2992   
16    \nCracking open the last closed platform: the ...         821   
17    \nThe COOLEST is a portable party disguised as...         162   
18    \nWasteland 2 has been released to great accla...        2916   
19    \nYou did it: you brought back MYSTERY SCIENCE...        2863   
20    \n"Hell, there are no rules here - we're tryin...        2063   
21    \nTRAVEL JACKET with built-in Neck Pillow, Eye...        3089   
22    \nPlanetary Annihilation brings RTS gameplay t...         760   
23    \nAn epic RPG with turn-based combat, cooperat...         755  

In [226]:
inventor_df = pd.DataFrame(inventor_list, columns = ['Inventor'])
count = 0
for i,each in enumerate(inventor_df['Inventor']):
    inventor_df.loc[i,'Inventor ID'] = int(count)
    count+=1

In [229]:
inventor_df= inventor_df[['Inventor ID', 'Inventor']]
inventor_df.head()

Inventor ID          Inventor
0          0.0   Two Tumbleweeds
1          1.0       Kill Screen
2          2.0     Katrina Spade
3          3.0    I AM Cardboard
4          4.0  John Szczepaniak

In [230]:
category_df = pd.DataFrame(category_list, columns = ['Category'])
count = 0
for i,each in enumerate(category_df['Category']):
    category_df.loc[i, 'Category ID'] = int(count)
    count+=1
category_df = category_df[['Category ID', 'Category']]
category_df.head()

Category ID         Category
0          0.0         Hardware
1          1.0            Dance
2          2.0           Crafts
3          3.0  Gaming Hardware
4          4.0   Conceptual Art

In [254]:
inventor_df.to_sql('Inventor', conn, if_exists='replace', index=False )

C:\Users\hashe\Anaconda3\lib\site-packages\pandas\core\generic.py:1362: UserWarning: The spaces in these column names will not be changed. In pandas versions < 0.14, spaces were converted to underscores.
  chunksize=chunksize, dtype=dtype)


In [232]:
category_df.to_sql('Category', conn, if_exists = 'replace', index=False)

C:\Users\hashe\Anaconda3\lib\site-packages\pandas\core\generic.py:1362: UserWarning: The spaces in these column names will not be changed. In pandas versions < 0.14, spaces were converted to underscores.
  chunksize=chunksize, dtype=dtype)


## The tables in our database:

In [256]:
run_query('Select name from sqlite_master where type = "table"')

name
0  Pledge_Tiers
1      Category
2   most_backed
3      Inventor

### We now have 4 tables in our 'kickstarter_prokects' database. Pledge tiers is a table with the various pledge tiers and its associated backers corresponding to each project. Category is a table containing the various categories that a project can fall under. Inventor is a table that contains the name of the creator/inventor for each project. The most_backed table consists of information for each of our kickstarter projects.

<img src = 'Table_Schema.png'>

# Let us now check whether each of our tables is in first normal form

(For the purpose of the use-cases clause of this assignment, i have created a separate table for inventor and category by creating artificial IDs for each category.)

In [257]:
run_query('Select * from most_backed limit 5')

Project ID  Amount Pledged  \
0           0       8782571.0   
1           1       6465690.0   
2           2       5408916.0   
3           3       5702153.0   
4           4       3336371.0   

                                               Blurb  Inventor ID  \
0  \nThis is a card game for people who are into ...          930   
1  \nAn unusually addicting, high-quality desk to...           92   
2  \nBring Reading Rainbow’s library of interacti...         1514   
3  \nUPDATED: This is it. We're making a Veronica...         1928   
4  \nAn adventure game from Tim Schafer, Double F...          627   

   Category ID Currency       Goal           Location  No. Backers  \
0          112      usd    10000.0    Los Angeles, CA       219382   
1          101      usd    15000.0         Denver, CO       154926   
2           64      usd  1000000.0    Los Angeles, CA       105857   
3           78      usd  2000000.0      San Diego, CA        91585   
4           47      usd   400000.0  San Francisco, CA        87142   

                                               Title  \
0                                  Exploding Kittens   
1                      Fidget Cube: A Vinyl Desk Toy   
2  Bring Reading Rainbow Back for Every Child, Ev...   
3                    The Veronica Mars Movie Project   
4                              Double Fine Adventure   

                                                 URL  
0                /projects/elanlee/exploding-kittens  
1   /projects/antsylabs/fidget-cube-a-vinyl-desk-toy  
2  /projects/readingrainbow/bring-reading-rainbow...  
3  /projects/559914737/the-veronica-mars-movie-pr...  
4         /projects/doublefine/double-fine-adventure

This table seems to be in the first normal form as: <br>
1 There is a primary key ( Project ID) <br>
2 There are no repeating groups <br>
3 The values in each column of a table are atomic 

In [263]:
run_query('select * from inventor limit 5')

Inventor ID          Inventor
0          0.0   Two Tumbleweeds
1          1.0       Kill Screen
2          2.0     Katrina Spade
3          3.0    I AM Cardboard
4          4.0  John Szczepaniak

In [264]:
run_query('select * from category limit 5')

Category ID         Category
0          0.0         Hardware
1          1.0            Dance
2          2.0           Crafts
3          3.0  Gaming Hardware
4          4.0   Conceptual Art

In [265]:
run_query('select * from pledge_tiers limit 5')

Pledge ID  Pledge Tiers   Backers
0          0          20.0   15505.0
1          0          35.0  202934.0
2          0         100.0     200.0
3          0         500.0       5.0
4          1           1.0     788.0

### Each of our tables is in first normal form.

## Use-cases:

### 1. List all the projects that are in the games category 

In [271]:
run_query('SELECT * FROM most_backed INNER JOIN category ON most_backed."Category ID" = category."Category ID" WHERE category.category LIKE "%Games%" ')

Project ID  Amount Pledged  \
0              4       3336371.0   
1              6       4188927.0   
2              7       3986929.0   
3              8       2090104.0   
4             10       6333295.0   
5             12       3845170.0   
6             14       5545991.0   
7             18       2933252.0   
8             22       2229344.0   
9             23       2032434.0   
10            24       2785537.0   
11            27       1836447.0   
12            28       1106371.0   
13            30       2134374.0   
14            31       1519680.0   
15            32       1229015.0   
16            33       1204726.0   
17            38        702516.0   
18            41       1578316.0   
19            43       2485506.0   
20            44        645158.0   
21            48        751920.0   
22            50       1919275.0   
23            51       1321306.0   
24            53       1538425.0   
25            55       1350700.0   
26            61        723886.0   
27            62       1578950.0   
28            64        944282.0   
29            67        776084.0   
...          ...             ...   
1274         977        101716.0   
1275        1208        108065.0   
1276        1387         72107.0   
1277        1407        728985.0   
1278        1459         82879.0   
1279        1708         93845.0   
1280        1786        380015.0   
1281        2036        239955.0   
1282        2062        130447.0   
1283        2142         71179.0   
1284        2161        100428.0   
1285        2606         39651.0   
1286        2636        354997.0   
1287        2669         62723.0   
1288        2793          4773.0   
1289        2899         49003.0   
1290        3003         78118.0   
1291        3015         77471.0   
1292        3088        207804.0   
1293        3103         66246.0   
1294        3205        109970.0   
1295        3216        120263.0   
1296        3231        233852.0   
1297        3284         47746.0   
1298        3547         30787.0   
1299        3673         80608.0   
1300        3768         26040.0   
1301        3854         50055.0   
1302        3863        117648.0   
1303        3893         96326.0   

                                                  Blurb  Inventor ID  \
0     \nAn adventure game from Tim Schafer, Double F...          627   
1     \nA story-driven CRPG set in the world of Mont...         2916   
2     \nPillars of Eternity is an isometric, party-b...          926   
3     \nThe creative talent behind Banjo-Kazooie and...          562   
4     \nYu Suzuki presents the long awaited third in...         2263   
5     \nClassic Japanese side-scrolling action, evol...          237   
6     \nIGAVANIA (n): A gothic, exploration-focused ...         2548   
7     \nWasteland 2 has been released to great accla...         2916   
8     \nPlanetary Annihilation brings RTS gameplay t...          760   
9     \nAn epic RPG with turn-based combat, cooperat...          755   
10    \nMissed the Kickstarter? Become a Late Reinfo...         2018   
11    \nShadowrun Returns brings back one of our mos...         2018   
12    \nRealistic single-player RPG set in the medie...         1389   
13    \nReclaim the stars in the exciting new Space ...          571   
14    \nFrom the creators of Wasteland 2 and Torment...         2916   
15    \nA tactical strategy game on an epic fantasy ...          627   
16    \nA Shadowrun cyberpunk cRPG set in 2056's Mag...         2018   
17    \nCastle Story is a voxel-based creative strat...          384   
18    \nTake a ship and 100 credits to make money le...         3086   
19    \nAn adventure game based on Homestuck, by And...         1474   
20    \nExplore a beautiful, vast and ruined world r...         1406   
21    \nA sandbox strategy game with town building, ...           30   
22    \nLord British returns to his fantasy RPG root...         1471   
23    \nAn all-new, immersive adventure game 

We see that an overwhelming majority (1304 out of 4000) of the most backed projects on kickstarter are in the games category ! (This includes Video Games, Tabletop Games etc)

### List all the inventors from Texas

In [274]:
run_query('SELECT inventor.inventor, most_backed.location FROM most_backed INNER JOIN inventor ON most_backed."inventor id" = inventor."inventor id" WHERE most_backed.location LIKE "%TX"')

Inventor         Location
0              Cyanide And Happiness       Dallas, TX
1                      Rooster Teeth       Austin, TX
2                  Portalarium, Inc.       Austin, TX
3                 Tinnus Enterprises        Plano, TX
4                              Stoic       Austin, TX
5                  Reaper Miniatures       Denton, TX
6       ArtCraft Entertainment, Inc.       Austin, TX
7                  Reaper Miniatures       Denton, TX
8             Kris, Rob, Matt & Dave       Dallas, TX
9                  Airship Syndicate       Austin, TX
10                      Replay Games       Austin, TX
11   The Fun Pimps Entertainment LLC       Dallas, TX
12                 Reaper Miniatures       Denton, TX
13                     Albino Dragon       Austin, TX
14                   BlankMediaGames       Austin, TX
15                            Knocki      Houston, TX
16                Descendent Studios       Austin, TX
17                   BlankMediaGames       Austin, TX
18                     BSX Athletics       Austin, TX
19                    Zachary Weiner      Houston, TX
20               Steve Jackson Games       Austin, TX
21                    Skyshine Games       Austin, TX
22                    Zachary Weiner      Houston, TX
23                             Ngozi       Austin, TX
24                        Solo Stove   Fort Worth, TX
25        Dan Provost & Tom Gerhardt       Austin, TX
26                    don hertzfeldt       Austin, TX
27                    Sandy Petersen     Rockwall, TX
28                    Sandy Petersen     Rockwall, TX
29         HOT Smart Watch from PHTL       Dallas, TX
..                               ...              ...
124                       Fwoosh LLC  San Antonio, TX
125                 Jackson Robinson         Kyle, TX
126                      Queen Games       Dallas, TX
127                    Albino Dragon       Austin, TX
128                    Foxtrot Games       Dallas, TX
129                       B&A Studio      Houston, TX
130                    Cobblebot LLC      Houston, TX
131                       B&A Studio      Houston, TX
132                       Dann Kriss      Kountze, TX
133                 Bobby@Go7 Gaming       Austin, TX
134                     Robin Arnott       Austin, TX
135                Charlie Brumfield     Mesquite, TX
136           Lisabeth Sewell McCann       Austin, TX
137                Jenn Giles Kemper         Waco, TX
138                         GlocalMe       Dallas, TX
139                   Emergent Games      Rowlett, TX
140                        Tim Doyle       Austin, TX
141                     Rebecca Gray       Austin, TX
142               Red Republic Games       Austin, TX
143                Christopher Locke       Austin, TX
144                      Queen Games       Dallas, TX
145                     Jeremy Snead       Dallas, TX
146                Charlie Brumfield     Mesquite, TX
147                 Azpen Innovation        Plano, TX
148                       eoseek llc       Austin, TX
149             Bombshell Miniatures     Mckinney, TX
150                      GREG PABICH       Keller, TX
151                   SBARK dog toys       Dallas, TX
152                   Nicholas Vitek      Houston, TX
153                         PulsePet       Dallas, TX

[154 rows x 2 columns]

There are 154 inventors from Texas

### 3. How many people pledged more than 500 ?

In [282]:
run_query('SELECT SUM(Backers) FROM pledge_tiers WHERE "Pledge tiers" > 500')

SUM(Backers)
0       89256.0

Close to 90,000 people have pledged more than 500$ !

### 4. Where do most of the inventors live ?

In [286]:
run_query('SELECT location, count(location) FROM most_backed GROUP BY location ORDER BY count(location) DESC LIMIT 5')

Location  count(location)
0  San Francisco, CA              280
1    Los Angeles, CA              271
2       New York, NY              212
3        Seattle, WA              147
4         London, UK              132

Unsurprisingly, most of the creators are based out of San Francisco, California followed by Los Angeles, CA

### 5. Which are the top 5 most funded projects ?

In [287]:
run_query('SELECT * FROM most_backed ORDER BY "Amount Pledged" DESC LIMIT 5')

Project ID  Amount Pledged  \
0           5      20338986.0   
1          17      13285226.0   
2          13      12779843.0   
3          11      10266845.0   
4          21       9192055.0   

                                               Blurb  Inventor ID  \
0  \nColor e-paper smartwatch with up to 7 days o...         1709   
1  \nThe COOLEST is a portable party disguised as...          162   
2  \nTwo affordable, heart rate-enabled smartwatc...         1709   
3  \nPebble is a customizable watch. Download new...         1709   
4  \nTRAVEL JACKET with built-in Neck Pillow, Eye...         3089   

   Category ID Currency       Goal          Location  No. Backers  \
0          101      usd   500000.0     Palo Alto, CA        78471   
1          101      usd    50000.0      Portland, OR        62642   
2          101      usd  1000000.0  Redwood City, CA        66673   
3          101      usd   100000.0     Palo Alto, CA        68929   
4          101      usd    20000.0       Chicago, IL        44949   

                                               Title  \
0   Pebble Time - Awesome Smartwatch, No Compromises   
1  COOLEST COOLER: 21st Century Cooler that's Act...   
2             Pebble 2, Time 2 + All-New Pebble Core   
3       Pebble: E-Paper Watch for iPhone and Android   
4  The World's Best TRAVEL JACKET with 15 Feature...   

                                                 URL  
0  /projects/597507018/pebble-time-awesome-smartw...  
1  /projects/ryangrepper/coolest-cooler-21st-cent...  
2  /projects/597507018/pebble-2-time-2-and-core-a...  
3  /projects/597507018/pebble-e-paper-watch-for-i...  
4  /projects/597538543/the-worlds-best-travel-jac...